addons/binding/org.openhab.binding.zoneminder/src/main/java/org/openhab/binding/zoneminder/handler/ZoneMinderServerBridgeHandler.java

```diff
@@ -50,69 +53,87 @@
+ * @author Martin S. Eskildsen - Initial contribution
  *
  */
-public class ZoneMinderServerBridgeHandler extends BaseBridgeHandler implements ZoneMinderHandler {
+public class ZoneMinderServerBridgeHandler extends BaseBridgeHandler implements IZoneMinderHandler {
 
     public static final int TELNET_TIMEOUT = 5000;
+    static final int HTTP_TIMEOUT = 5000;
 
     public static final Set<ThingTypeUID> SUPPORTED_THING_TYPES = Sets
             .newHashSet(ZoneMinderConstants.THING_TYPE_BRIDGE_ZONEMINDER_SERVER);
 
     /**
      * Logger
      */
-    private final Logger logger = LoggerFactory.getLogger(getClass());
+    private Logger logger = LoggerFactory.getLogger(getClass());
+
+    private ZoneMinderConnectionStatus zmConnectStatus = ZoneMinderConnectionStatus.UNINITIALIZED;
+    private ZoneMinderConnectionStatus lastSucceededStatus = ZoneMinderConnectionStatus.UNINITIALIZED;
 
-    private ZoneMinderDiscoveryService discoveryService = null;
-    private ServiceRegistration discoveryRegistration = null;
+    private RefreshPriority forcedPriority = RefreshPriority.UNKNOWN;
 
-    private ScheduledFuture<?> taskWatchDog = null;
-    private int refreshFrequency = 0;
-    private int refreshCycleCount = 0;
+    private ZoneMinderDiscoveryService discoveryService;
+    private ServiceRegistration discoveryRegistration;
+
+    private ScheduledFuture<?> taskWatchDog;
+    private Integer refreshCycleCount = 0;
 
     /** Connection status for the bridge. */
     private boolean connected = false;
-    private ThingStatus curBridgeStatus = ThingStatus.UNKNOWN;
-
-    protected boolean _online = false;
 
     private Runnable watchDogRunnable = new Runnable() {
-        private int watchDogCount = -1;
+        private int watchDogCount = 0;
 
         @Override
         public void run() {
-
             try {
-                updateAvaliabilityStatus(zoneMinderConnection);
+                updateAvaliabilityStatus(getZoneMinderConnection());
+
+                // Only Discover if Bridge is online
+                if (thing.getStatusInfo().getStatus() != ThingStatus.ONLINE) {
+                    return;
+                }
 
-                if ((discoveryService != null) && (getBridgeConfig().getAutodiscoverThings() == true)) {
+                // Check if autodiscovery is enabled
+                boolean bAutoDiscover = getBridgeConfig().getAutodiscoverThings();
+
+                if ((discoveryService != null) && bAutoDiscover) {
                     watchDogCount++;
                     // Run every two minutes
                     if ((watchDogCount % 8) == 0) {
                         discoveryService.startBackgroundDiscovery();
                         watchDogCount = 0;
                     }
                 }
+
             } catch (Exception exception) {
-                logger.error("[WATCHDOG]: Server run(): Exception: {}", exception.getMessage());
+                StackTraceElement ste = exception.getStackTrace()[0];
```

Comment:
What are you doing here, do you really need to inspect the stacktrace, does it happen often or only incidental?